### ISSUE TITLE
Is there a way to tell perceval to download metadata only?

### ISSUE THREAD

**Issue by jbon** [OPEN-NEXT/wp2.2_dev#33]
It seems that the command `git = Git(repo_URL, data_dump_path)` in `get_commits.py:51` downloads all the contents from a repo, not only the metadata, but also the files. It packs everything in *.pack files in the directory <reponame>/objects/pack. For hardware projects, this is a lot of data. And since we fetch infos not only from root repos but also from forks, it is really a lot of data. It is more than what my system can handle and makre the whole process really slow, way too slow for the reactivity that would be required from a management dashboard. Because of this, I can't test our algorithms on the really interesting projects who have a lot of data. This is a real handicap.

Can we simply tell perceval to skip downloading/generating the *.pack files? 

As an illustration, here the result of the `du` command in the folder `__DATA__/grimoire_dumps` on my machine. It contains Gb of data I don't use. Just for the repo AngelLM/Thor, these are **2.6 Gb**, and there are **150** forks of this repo, so processing this repo would mean downloading **360 Gb**, which is more than the total capacity of my disk (and an order of magnitude more than the space allocated to my linux distribution)!

```
jb2971@VirtualBox:~/GitRepos/wp2.2_dev/__DATA__/grimoire_dumps$ du
4	./rafaellaantoniou-wp2.2_reference/branches
4	./rafaellaantoniou-wp2.2_reference/refs/heads
4	./rafaellaantoniou-wp2.2_reference/refs/tags
12	./rafaellaantoniou-wp2.2_reference/refs
8	./rafaellaantoniou-wp2.2_reference/info
56	./rafaellaantoniou-wp2.2_reference/hooks
8	./rafaellaantoniou-wp2.2_reference/objects/d4
8	./rafaellaantoniou-wp2.2_reference/objects/8f
8	./rafaellaantoniou-wp2.2_reference/objects/56
8	./rafaellaantoniou-wp2.2_reference/objects/5a
12	./rafaellaantoniou-wp2.2_reference/objects/ec
4	./rafaellaantoniou-wp2.2_reference/objects/info
12	./rafaellaantoniou-wp2.2_reference/objects/06
8	./rafaellaantoniou-wp2.2_reference/objects/66
8	./rafaellaantoniou-wp2.2_reference/objects/f9
8	./rafaellaantoniou-wp2.2_reference/objects/1b
8	./rafaellaantoniou-wp2.2_reference/objects/ea
8	./rafaellaantoniou-wp2.2_reference/objects/1f
8	./rafaellaantoniou-wp2.2_reference/objects/a3
8	./rafaellaantoniou-wp2.2_reference/objects/4e
8	./rafaellaantoniou-wp2.2_reference/objects/da
8	./rafaellaantoniou-wp2.2_reference/objects/d3
8	./rafaellaantoniou-wp2.2_reference/objects/5d
8	./rafaellaantoniou-wp2.2_reference/objects/b5
8	./rafaellaantoniou-wp2.2_reference/objects/0a
8	./rafaellaantoniou-wp2.2_reference/objects/53
8	./rafaellaantoniou-wp2.2_reference/objects/02
8	./rafaellaantoniou-wp2.2_reference/objects/27
8	./rafaellaantoniou-wp2.2_reference/objects/9d
8	./rafaellaantoniou-wp2.2_reference/objects/3d
8	./rafaellaantoniou-wp2.2_reference/objects/79
8	./rafaellaantoniou-wp2.2_reference/objects/12
8	./rafaellaantoniou-wp2.2_reference/objects/4f
8	./rafaellaantoniou-wp2.2_reference/objects/5f
12	./rafaellaantoniou-wp2.2_reference/objects/65
12	./rafaellaantoniou-wp2.2_reference/objects/b2
8	./rafaellaantoniou-wp2.2_reference/objects/a8
8	./rafaellaantoniou-wp2.2_reference/objects/a1
8	./rafaellaantoniou-wp2.2_reference/objects/18
8	./rafaellaantoniou-wp2.2_reference/objects/1d
8	./rafaellaantoniou-wp2.2_reference/objects/34
8	./rafaellaantoniou-wp2.2_reference/objects/74
8	./rafaellaantoniou-wp2.2_reference/objects/4a
8	./rafaellaantoniou-wp2.2_reference/objects/0f
8	./rafaellaantoniou-wp2.2_reference/objects/f5
8	./rafaellaantoniou-wp2.2_reference/objects/86
8	./rafaellaantoniou-wp2.2_reference/objects/9c
12	./rafaellaantoniou-wp2.2_reference/objects/3c
8	./rafaellaantoniou-wp2.2_reference/objects/3b
8	./rafaellaantoniou-wp2.2_reference/objects/ff
8	./rafaellaantoniou-wp2.2_reference/objects/38
12	./rafaellaantoniou-wp2.2_reference/objects/81
8	./rafaellaantoniou-wp2.2_reference/objects/67
12	./rafaellaantoniou-wp2.2_reference/objects/4d
8	./rafaellaantoniou-wp2.2_reference/objects/b6
12	./rafaellaantoniou-wp2.2_reference/objects/5c
8	./rafaellaantoniou-wp2.2_reference/objects/8e
8	./rafaellaantoniou-wp2.2_reference/objects/32
12	./rafaellaantoniou-wp2.2_reference/objects/92
12	./rafaellaantoniou-wp2.2_reference/objects/51
8	./rafaellaantoniou-wp2.2_reference/objects/ac
20	./rafaellaantoniou-wp2.2_reference/objects/f2
8	./rafaellaantoniou-wp2.2_reference/objects/28
12	./rafaellaantoniou-wp2.2_reference/objects/e9
8	./rafaellaantoniou-wp2.2_reference/objects/7f
8	./rafaellaantoniou-wp2.2_reference/objects/52
4	./rafaellaantoniou-wp2.2_reference/objects/pack
8	./rafaellaantoniou-wp2.2_reference/objects/bf
8	./rafaellaantoniou-wp2.2_reference/objects/a0
8	./rafaellaantoniou-wp2.2_reference/objects/dc
8	./rafaellaantoniou-wp2.2_reference/objects/80
8	./rafaellaantoniou-wp2.2_reference/objects/a2
8	./rafaellaantoniou-wp2.2_reference/objects/d1
8	./rafaellaantoniou-wp2.2_reference/objects/47
12	./rafaellaantoniou-wp2.2_reference/objects/21
8	./rafaellaantoniou-wp2.2_reference/objects/4c
8	./rafaellaantoniou-wp2.2_reference/objects/99
8	./rafaellaantoniou-wp2.2_reference/objects/48
8	./rafaellaantoniou-wp2.2_reference/objects/91
12	./rafaellaantoniou-wp2.2_reference/objects/46
8	./rafaellaantoniou-wp2.2_reference/objects/0e
8	./rafaellaantoniou-wp2.2_reference/objects/35
8	./rafaellaantoniou-wp2.2_reference/objects/2a
8	./rafaellaantoniou-wp2.2_reference/objects/7a
8	./rafaellaantoniou-wp2.2_reference/objects/af
8	./rafaellaantoniou-wp2.2_reference/objects/6b
8	./rafaellaantoniou-wp2.2_reference/objects/7d
708	./rafaellaantoniou-wp2.2_reference/objects
812	./rafaellaantoniou-wp2.2_reference
4	./jurgenwesterhoff-bGeigieNanoKit/branches
4	./jurgenwesterhoff-bGeigieNanoKit/refs/heads
4	./jurgenwesterhoff-bGeigieNanoKit/refs/tags
12	./jurgenwesterhoff-bGeigieNanoKit/refs
8	./jurgenwesterhoff-bGeigieNanoKit/info
56	./jurgenwesterhoff-bGeigieNanoKit/hooks
4	./jurgenwesterhoff-bGeigieNanoKit/objects/info
64380	./jurgenwesterhoff-bGeigieNanoKit/objects/pack
64388	./jurgenwesterhoff-bGeigieNanoKit/objects
64492	./jurgenwesterhoff-bGeigieNanoKit
4	./OPEN-NEXT-wp2.2_reference/branches
4	./OPEN-NEXT-wp2.2_reference/refs/heads
4	./OPEN-NEXT-wp2.2_reference/refs/tags
12	./OPEN-NEXT-wp2.2_reference/refs
8	./OPEN-NEXT-wp2.2_reference/info
56	./OPEN-NEXT-wp2.2_reference/hooks
8	./OPEN-NEXT-wp2.2_reference/objects/d4
8	./OPEN-NEXT-wp2.2_reference/objects/8f
8	./OPEN-NEXT-wp2.2_reference/objects/56
8	./OPEN-NEXT-wp2.2_reference/objects/5a
12	./OPEN-NEXT-wp2.2_reference/objects/ec
4	./OPEN-NEXT-wp2.2_reference/objects/info
12	./OPEN-NEXT-wp2.2_reference/objects/06
8	./OPEN-NEXT-wp2.2_reference/objects/66
8	./OPEN-NEXT-wp2.2_reference/objects/f9
8	./OPEN-NEXT-wp2.2_reference/objects/1b
8	./OPEN-NEXT-wp2.2_reference/objects/ea
8	./OPEN-NEXT-wp2.2_reference/objects/1f
8	./OPEN-NEXT-wp2.2_reference/objects/a3
8	./OPEN-NEXT-wp2.2_reference/objects/4e
8	./OPEN-NEXT-wp2.2_reference/objects/da
8	./OPEN-NEXT-wp2.2_reference/objects/d3
8	./OPEN-NEXT-wp2.2_reference/objects/5d
8	./OPEN-NEXT-wp2.2_reference/objects/b5
8	./OPEN-NEXT-wp2.2_reference/objects/0a
8	./OPEN-NEXT-wp2.2_reference/objects/53
8	./OPEN-NEXT-wp2.2_reference/objects/02
8	./OPEN-NEXT-wp2.2_reference/objects/27
8	./OPEN-NEXT-wp2.2_reference/objects/9d
8	./OPEN-NEXT-wp2.2_reference/objects/3d
8	./OPEN-NEXT-wp2.2_reference/objects/79
8	./OPEN-NEXT-wp2.2_reference/objects/12
8	./OPEN-NEXT-wp2.2_reference/objects/4f
8	./OPEN-NEXT-wp2.2_reference/objects/5f
12	./OPEN-NEXT-wp2.2_reference/objects/65
12	./OPEN-NEXT-wp2.2_reference/objects/b2
8	./OPEN-NEXT-wp2.2_reference/objects/a8
8	./OPEN-NEXT-wp2.2_reference/objects/a1
8	./OPEN-NEXT-wp2.2_reference/objects/18
8	./OPEN-NEXT-wp2.2_reference/objects/1d
8	./OPEN-NEXT-wp2.2_reference/objects/34
8	./OPEN-NEXT-wp2.2_reference/objects/74
8	./OPEN-NEXT-wp2.2_reference/objects/4a
8	./OPEN-NEXT-wp2.2_reference/objects/0f
8	./OPEN-NEXT-wp2.2_reference/objects/f5
8	./OPEN-NEXT-wp2.2_reference/objects/86
8	./OPEN-NEXT-wp2.2_reference/objects/9c
12	./OPEN-NEXT-wp2.2_reference/objects/3c
8	./OPEN-NEXT-wp2.2_reference/objects/61
8	./OPEN-NEXT-wp2.2_reference/objects/3b
8	./OPEN-NEXT-wp2.2_reference/objects/ff
8	./OPEN-NEXT-wp2.2_reference/objects/38
12	./OPEN-NEXT-wp2.2_reference/objects/81
8	./OPEN-NEXT-wp2.2_reference/objects/67
12	./OPEN-NEXT-wp2.2_reference/objects/4d
8	./OPEN-NEXT-wp2.2_reference/objects/b6
12	./OPEN-NEXT-wp2.2_reference/objects/5c
8	./OPEN-NEXT-wp2.2_reference/objects/8e
8	./OPEN-NEXT-wp2.2_reference/objects/32
12	./OPEN-NEXT-wp2.2_reference/objects/92
12	./OPEN-NEXT-wp2.2_reference/objects/51
8	./OPEN-NEXT-wp2.2_reference/objects/ac
20	./OPEN-NEXT-wp2.2_reference/objects/f2
8	./OPEN-NEXT-wp2.2_reference/objects/28
12	./OPEN-NEXT-wp2.2_reference/objects/e9
8	./OPEN-NEXT-wp2.2_reference/objects/7f
8	./OPEN-NEXT-wp2.2_reference/objects/52
4	./OPEN-NEXT-wp2.2_reference/objects/pack
8	./OPEN-NEXT-wp2.2_reference/objects/bf
8	./OPEN-NEXT-wp2.2_reference/objects/a0
8	./OPEN-NEXT-wp2.2_reference/objects/dc
8	./OPEN-NEXT-wp2.2_reference/objects/80
8	./OPEN-NEXT-wp2.2_reference/objects/a2
8	./OPEN-NEXT-wp2.2_reference/objects/d1
8	./OPEN-NEXT-wp2.2_reference/objects/47
12	./OPEN-NEXT-wp2.2_reference/objects/21
8	./OPEN-NEXT-wp2.2_reference/objects/4c
8	./OPEN-NEXT-wp2.2_reference/objects/99
8	./OPEN-NEXT-wp2.2_reference/objects/48
8	./OPEN-NEXT-wp2.2_reference/objects/91
12	./OPEN-NEXT-wp2.2_reference/objects/46
8	./OPEN-NEXT-wp2.2_reference/objects/0e
8	./OPEN-NEXT-wp2.2_reference/objects/35
8	./OPEN-NEXT-wp2.2_reference/objects/2a
8	./OPEN-NEXT-wp2.2_reference/objects/7a
8	./OPEN-NEXT-wp2.2_reference/objects/af
8	./OPEN-NEXT-wp2.2_reference/objects/6b
8	./OPEN-NEXT-wp2.2_reference/objects/7d
716	./OPEN-NEXT-wp2.2_reference/objects
820	./OPEN-NEXT-wp2.2_reference
4	./OPEN-NEXT-WP2.2_dev/branches
4	./OPEN-NEXT-WP2.2_dev/refs/heads
4	./OPEN-NEXT-WP2.2_dev/refs/tags
12	./OPEN-NEXT-WP2.2_dev/refs
8	./OPEN-NEXT-WP2.2_dev/info
56	./OPEN-NEXT-WP2.2_dev/hooks
4	./OPEN-NEXT-WP2.2_dev/objects/info
552	./OPEN-NEXT-WP2.2_dev/objects/pack
560	./OPEN-NEXT-WP2.2_dev/objects
664	./OPEN-NEXT-WP2.2_dev
4	./penyuan-github-mining/branches
4	./penyuan-github-mining/refs/heads
4	./penyuan-github-mining/refs/tags
12	./penyuan-github-mining/refs
8	./penyuan-github-mining/info
56	./penyuan-github-mining/hooks
4	./penyuan-github-mining/objects/info
8868	./penyuan-github-mining/objects/pack
8876	./penyuan-github-mining/objects
8980	./penyuan-github-mining
4	./massjona-github-mining/branches
4	./massjona-github-mining/refs/heads
4	./massjona-github-mining/refs/tags
12	./massjona-github-mining/refs
8	./massjona-github-mining/info
56	./massjona-github-mining/hooks
8	./massjona-github-mining/objects/8f
4	./massjona-github-mining/objects/info
8	./massjona-github-mining/objects/66
8	./massjona-github-mining/objects/90
8	./massjona-github-mining/objects/31
8	./massjona-github-mining/objects/a3
12	./massjona-github-mining/objects/eb
8	./massjona-github-mining/objects/78
8	./massjona-github-mining/objects/d3
8	./massjona-github-mining/objects/be
8	./massjona-github-mining/objects/f6
12	./massjona-github-mining/objects/3d
8	./massjona-github-mining/objects/c5
8	./massjona-github-mining/objects/65
8	./massjona-github-mining/objects/b1
20	./massjona-github-mining/objects/1e
8	./massjona-github-mining/objects/2b
8	./massjona-github-mining/objects/e5
8	./massjona-github-mining/objects/61
12	./massjona-github-mining/objects/f4
8	./massjona-github-mining/objects/85
8	./massjona-github-mining/objects/81
8	./massjona-github-mining/objects/43
8	./massjona-github-mining/objects/01
12	./massjona-github-mining/objects/b6
8	./massjona-github-mining/objects/98
12	./massjona-github-mining/objects/03
20	./massjona-github-mining/objects/94
8	./massjona-github-mining/objects/92
8	./massjona-github-mining/objects/2f
8	./massjona-github-mining/objects/28
12	./massjona-github-mining/objects/52
4	./massjona-github-mining/objects/pack
8	./massjona-github-mining/objects/04
8	./massjona-github-mining/objects/e4
8	./massjona-github-mining/objects/29
8	./massjona-github-mining/objects/88
8	./massjona-github-mining/objects/80
8	./massjona-github-mining/objects/25
12	./massjona-github-mining/objects/e3
8	./massjona-github-mining/objects/1a
8	./massjona-github-mining/objects/c9
8	./massjona-github-mining/objects/7e
8	./massjona-github-mining/objects/48
12	./massjona-github-mining/objects/db
8	./massjona-github-mining/objects/46
8	./massjona-github-mining/objects/e0
8	./massjona-github-mining/objects/14
8	./massjona-github-mining/objects/6b
8	./massjona-github-mining/objects/fe
8	./massjona-github-mining/objects/75
8	./massjona-github-mining/objects/ba
8	./massjona-github-mining/objects/9a
476	./massjona-github-mining/objects
580	./massjona-github-mining
4	./AngelLM-Thor/branches
4	./AngelLM-Thor/refs/heads
4	./AngelLM-Thor/refs/tags
12	./AngelLM-Thor/refs
8	./AngelLM-Thor/info
56	./AngelLM-Thor/hooks
4	./AngelLM-Thor/objects/info
2605124	./AngelLM-Thor/objects/pack
2605132	./AngelLM-Thor/objects
2605232	./AngelLM-Thor
4	./OPEN-NEXT-wp2.2_dev/branches
28	./OPEN-NEXT-wp2.2_dev/refs/heads
4	./OPEN-NEXT-wp2.2_dev/refs/tags
36	./OPEN-NEXT-wp2.2_dev/refs
8	./OPEN-NEXT-wp2.2_dev/info
56	./OPEN-NEXT-wp2.2_dev/hooks
8	./OPEN-NEXT-wp2.2_dev/objects/24
8	./OPEN-NEXT-wp2.2_dev/objects/cb
4	./OPEN-NEXT-wp2.2_dev/objects/info
8	./OPEN-NEXT-wp2.2_dev/objects/06
8	./OPEN-NEXT-wp2.2_dev/objects/41
8	./OPEN-NEXT-wp2.2_dev/objects/5d
8	./OPEN-NEXT-wp2.2_dev/objects/2b
8	./OPEN-NEXT-wp2.2_dev/objects/13
8	./OPEN-NEXT-wp2.2_dev/objects/7c
8	./OPEN-NEXT-wp2.2_dev/objects/01
8	./OPEN-NEXT-wp2.2_dev/objects/c6
8	./OPEN-NEXT-wp2.2_dev/objects/10
8	./OPEN-NEXT-wp2.2_dev/objects/67
8	./OPEN-NEXT-wp2.2_dev/objects/de
8	./OPEN-NEXT-wp2.2_dev/objects/5b
576	./OPEN-NEXT-wp2.2_dev/objects/pack
8	./OPEN-NEXT-wp2.2_dev/objects/25
8	./OPEN-NEXT-wp2.2_dev/objects/99
8	./OPEN-NEXT-wp2.2_dev/objects/26
8	./OPEN-NEXT-wp2.2_dev/objects/fc
728	./OPEN-NEXT-wp2.2_dev/objects
856	./OPEN-NEXT-wp2.2_dev
4	./rafaellaantoniou-github-mining/branches
4	./rafaellaantoniou-github-mining/refs/heads
4	./rafaellaantoniou-github-mining/refs/tags
12	./rafaellaantoniou-github-mining/refs
8	./rafaellaantoniou-github-mining/info
56	./rafaellaantoniou-github-mining/hooks
4	./rafaellaantoniou-github-mining/objects/info
8484	./rafaellaantoniou-github-mining/objects/pack
8492	./rafaellaantoniou-github-mining/objects
8596	./rafaellaantoniou-github-mining
4	./jbon-wp2.2_reference/branches
4	./jbon-wp2.2_reference/refs/heads
4	./jbon-wp2.2_reference/refs/tags
12	./jbon-wp2.2_reference/refs
8	./jbon-wp2.2_reference/info
56	./jbon-wp2.2_reference/hooks
4	./jbon-wp2.2_reference/objects/info
36	./jbon-wp2.2_reference/objects/pack
44	./jbon-wp2.2_reference/objects
148	./jbon-wp2.2_reference
4	./Safecast-bGeigieNanoKit/branches
4	./Safecast-bGeigieNanoKit/refs/heads
4	./Safecast-bGeigieNanoKit/refs/tags
12	./Safecast-bGeigieNanoKit/refs
8	./Safecast-bGeigieNanoKit/info
56	./Safecast-bGeigieNanoKit/hooks
4	./Safecast-bGeigieNanoKit/objects/info
**128428**	./Safecast-bGeigieNanoKit/objects/pack
128436	./Safecast-bGeigieNanoKit/objects
128540	./Safecast-bGeigieNanoKit
4	./OPEN-NEXT-WP2.2_reference/branches
4	./OPEN-NEXT-WP2.2_reference/refs/heads
4	./OPEN-NEXT-WP2.2_reference/refs/tags
12	./OPEN-NEXT-WP2.2_reference/refs
8	./OPEN-NEXT-WP2.2_reference/info
56	./OPEN-NEXT-WP2.2_reference/hooks
8	./OPEN-NEXT-WP2.2_reference/objects/d4
8	./OPEN-NEXT-WP2.2_reference/objects/8f
8	./OPEN-NEXT-WP2.2_reference/objects/56
8	./OPEN-NEXT-WP2.2_reference/objects/5a
12	./OPEN-NEXT-WP2.2_reference/objects/ec
4	./OPEN-NEXT-WP2.2_reference/objects/info
12	./OPEN-NEXT-WP2.2_reference/objects/06
8	./OPEN-NEXT-WP2.2_reference/objects/66
8	./OPEN-NEXT-WP2.2_reference/objects/f9
8	./OPEN-NEXT-WP2.2_reference/objects/1b
8	./OPEN-NEXT-WP2.2_reference/objects/ea
8	./OPEN-NEXT-WP2.2_reference/objects/1f
8	./OPEN-NEXT-WP2.2_reference/objects/a3
8	./OPEN-NEXT-WP2.2_reference/objects/4e
8	./OPEN-NEXT-WP2.2_reference/objects/da
8	./OPEN-NEXT-WP2.2_reference/objects/d3
8	./OPEN-NEXT-WP2.2_reference/objects/5d
8	./OPEN-NEXT-WP2.2_reference/objects/b5
8	./OPEN-NEXT-WP2.2_reference/objects/0a
8	./OPEN-NEXT-WP2.2_reference/objects/53
8	./OPEN-NEXT-WP2.2_reference/objects/02
8	./OPEN-NEXT-WP2.2_reference/objects/27
8	./OPEN-NEXT-WP2.2_reference/objects/9d
8	./OPEN-NEXT-WP2.2_reference/objects/3d
8	./OPEN-NEXT-WP2.2_reference/objects/79
8	./OPEN-NEXT-WP2.2_reference/objects/12
8	./OPEN-NEXT-WP2.2_reference/objects/4f
8	./OPEN-NEXT-WP2.2_reference/objects/5f
12	./OPEN-NEXT-WP2.2_reference/objects/65
12	./OPEN-NEXT-WP2.2_reference/objects/b2
8	./OPEN-NEXT-WP2.2_reference/objects/a8
8	./OPEN-NEXT-WP2.2_reference/objects/a1
8	./OPEN-NEXT-WP2.2_reference/objects/18
8	./OPEN-NEXT-WP2.2_reference/objects/1d
8	./OPEN-NEXT-WP2.2_reference/objects/34
8	./OPEN-NEXT-WP2.2_reference/objects/74
8	./OPEN-NEXT-WP2.2_reference/objects/4a
8	./OPEN-NEXT-WP2.2_reference/objects/0f
8	./OPEN-NEXT-WP2.2_reference/objects/f5
8	./OPEN-NEXT-WP2.2_reference/objects/86
8	./OPEN-NEXT-WP2.2_reference/objects/9c
12	./OPEN-NEXT-WP2.2_reference/objects/3c
8	./OPEN-NEXT-WP2.2_reference/objects/61
8	./OPEN-NEXT-WP2.2_reference/objects/3b
8	./OPEN-NEXT-WP2.2_reference/objects/ff
8	./OPEN-NEXT-WP2.2_reference/objects/38
12	./OPEN-NEXT-WP2.2_reference/objects/81
8	./OPEN-NEXT-WP2.2_reference/objects/67
12	./OPEN-NEXT-WP2.2_reference/objects/4d
8	./OPEN-NEXT-WP2.2_reference/objects/b6
12	./OPEN-NEXT-WP2.2_reference/objects/5c
8	./OPEN-NEXT-WP2.2_reference/objects/8e
8	./OPEN-NEXT-WP2.2_reference/objects/32
12	./OPEN-NEXT-WP2.2_reference/objects/92
12	./OPEN-NEXT-WP2.2_reference/objects/51
8	./OPEN-NEXT-WP2.2_reference/objects/ac
20	./OPEN-NEXT-WP2.2_reference/objects/f2
8	./OPEN-NEXT-WP2.2_reference/objects/28
12	./OPEN-NEXT-WP2.2_reference/objects/e9
8	./OPEN-NEXT-WP2.2_reference/objects/7f
8	./OPEN-NEXT-WP2.2_reference/objects/52
4	./OPEN-NEXT-WP2.2_reference/objects/pack
8	./OPEN-NEXT-WP2.2_reference/objects/bf
8	./OPEN-NEXT-WP2.2_reference/objects/a0
8	./OPEN-NEXT-WP2.2_reference/objects/dc
8	./OPEN-NEXT-WP2.2_reference/objects/80
8	./OPEN-NEXT-WP2.2_reference/objects/a2
8	./OPEN-NEXT-WP2.2_reference/objects/d1
8	./OPEN-NEXT-WP2.2_reference/objects/47
12	./OPEN-NEXT-WP2.2_reference/objects/21
8	./OPEN-NEXT-WP2.2_reference/objects/4c
8	./OPEN-NEXT-WP2.2_reference/objects/99
8	./OPEN-NEXT-WP2.2_reference/objects/48
8	./OPEN-NEXT-WP2.2_reference/objects/91
12	./OPEN-NEXT-WP2.2_reference/objects/46
8	./OPEN-NEXT-WP2.2_reference/objects/0e
8	./OPEN-NEXT-WP2.2_reference/objects/35
8	./OPEN-NEXT-WP2.2_reference/objects/2a
8	./OPEN-NEXT-WP2.2_reference/objects/7a
8	./OPEN-NEXT-WP2.2_reference/objects/af
8	./OPEN-NEXT-WP2.2_reference/objects/6b
8	./OPEN-NEXT-WP2.2_reference/objects/7d
716	./OPEN-NEXT-WP2.2_reference/objects
820	./OPEN-NEXT-WP2.2_reference
4	./Warukira-bGeigieNanoKit/branches
4	./Warukira-bGeigieNanoKit/refs/heads
4	./Warukira-bGeigieNanoKit/refs/tags
12	./Warukira-bGeigieNanoKit/refs
8	./Warukira-bGeigieNanoKit/info
56	./Warukira-bGeigieNanoKit/hooks
4	./Warukira-bGeigieNanoKit/objects/info
128428	./Warukira-bGeigieNanoKit/objects/pack
128436	./Warukira-bGeigieNanoKit/objects
128536	./Warukira-bGeigieNanoKit
4	./jbon-github-mining/branches
4	./jbon-github-mining/refs/heads
4	./jbon-github-mining/refs/tags
12	./jbon-github-mining/refs
8	./jbon-github-mining/info
56	./jbon-github-mining/hooks
4	./jbon-github-mining/objects/info
8864	./jbon-github-mining/objects/pack
8872	./jbon-github-mining/objects
8976	./jbon-github-mining
2958056	.
```

**Comment by penyuan** [comment]
A couple observations:

1. It is the `perceval.backends.core.git` module that clones an entire git repository. The Perceval GitHub backend does not.
2. For some reason, running `git clone` from the command line does the huge download that @jbon described, but if I go to the GitHub page of the same repository and click on the webpage's download link, it gives me a much smaller file that's "just" a few hundred MBs instead of many GBs.

I am continuing this investigation with haste.

**Comment by jbon** [comment]
> It seems that the command `git = Git(repo_URL, data_dump_path)` in `get_commits.py:51` downloads all the contents from a repo, not only the metadata, but also the files.

data is actually downloaded by `git.fetch()` in `get_commits.py:59`, to be precise.

**Comment by jbon** [comment]
Created a (very ugly) bandaid in branch `iss33_bandaid`. It deletes the subfolder `objects/pack` for each cloned repo. But it is still not practicable because of the time perceval needs to download the (useless) data.

Another possible bandaid would be to renounce getting all commits of all forks until we found a solution for this issue.

**Comment by penyuan** [comment]
In this comment, I note the things we discussed/decided on during our meeting on 2020-07-17.

* It is impractical for us to use Perceval in its current state.
  * As confirmed by my emails with its developers, there currently isn't a way to directly tell Perceval not to fetch the whole repository. This is because Perceval uses the command line git client in the background for this purpose (via the `git clone` command).
* There are some Python git libraries:
  * [Dulwich](dulwich.io/)
  * [pygit2](https://www.pygit2.org/)
    * Technically, this library allows cloning with the `mirror` option which can reduce download sizes.
  * There are also some GitHub libraries but they wouldn't be a generic git solution.
* We can migrate to using GitHub's API version 4 which uses GraphQL queries.
  * I got in touch with a GitHub developer who has answered many questions, such as how branches *really* work.
  * This is a newer API and unlikely to be deprecated any time soon when compared to the previous REST version 3.
  * Familiarising us with GraphQL will also help interface with the Wikifactory API.
  * With GraphQL, we query for the exact data we need saving computer and brain power.

The decision - for now - is to give the GitHub GraphQL API a try. I will do this by attempting to recreate a commit history graph by pulling data via the API.

BTW, it's ideal if we can eventually contribute some of our code to upstream Perceval. To that end, we will stay on Python 3.6 for now because that's what Perceval uses, but I will ask them if they intend to move to newer Python versions soon.

**Comment by penyuan** [comment]
We've decided to migrate from Perceval to direct API calls to GitHub for metadata, because Perceval can't be told to not download so much data at this time. Their developers are interested in adding that flexibility but it doesn't look imminent.

1. I've tried a bit and we need a mixture of v3 and v4 API calls to get the information we want.
2. Since we're not using Perceval for now. We can move to the latest Python version (3.8.x at time of writing).
